bert_squeeze/data/modules/transformer_module.py

```diff
@@ -180,3 +178,132 @@ def featurize(self) -> datasets.DatasetDict:
+
+        self.train_batch_size = kwargs.get("train_batch_size", 32)
+        self.eval_batch_size = kwargs.get("eval_batch_size", 32)
+
+        self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+        self.dataset = None
+        self.train = None
+        self.test = None
+        self.val = None
+
+    def featurize(self) -> datasets.DatasetDict:
+        """
+        Returns:
+            DatasetDict: featurized dataset
+        """
+        tokenized_dataset = self.dataset.map(
+            lambda x: self.tokenizer(
+                x[self.source_col],
+                padding="longest",
+                max_length=self.max_source_length,
+                truncation=True,
+            )
+        )
+        with self.tokenizer.as_target_tokenizer():
+            tokenized_dataset = tokenized_dataset.map(
+                lambda x: {
+                    "labels": self.tokenizer(
+                        x[self.target_col],
+                        padding="longest",
+                        max_length=self.max_target_length,
+                        truncation=True,
+                    )["input_ids"]
+                }
+            )
+        columns = ["input_ids", "attention_mask", "labels"]
+        if not any(
+            [
+                model_name in self.tokenizer.name_or_path
+                for model_name in ["distilbert", "t5"]
+            ]
+        ):
+            columns += ["token_type_ids"]
+
+        tokenized_dataset.set_format(type='torch', columns=columns)
+        return tokenized_dataset
+
+    def setup(self, stage: Optional[str] = None):
+        """"""
+        featurized_dataset = self.featurize()
+        self.train = featurized_dataset["train"]
+        self.val = featurized_dataset["validation"]
+        self.test = featurized_dataset["test"]
+
+    def train_dataloader(self) -> DataLoader:
+        """
+        Returns:
+            DataLoader: train dataloader
+        """
+        return DataLoader(
+            self.train,
+            collate_fn=self._collate_fn(),
+            batch_size=self.train_batch_size,
+            drop_last=True,
+        )
+
+    def test_dataloader(self) -> DataLoader:
+        """
+        Returns:
+            DataLoader: test dataloader
+        """
+        return DataLoader(
+            self.test,
+            collate_fn=self._collate_fn(),
+            batch_size=self.eval_batch_size,
+            drop_last=True,
+        )
+
+    def val_dataloader(self) -> DataLoader:
+        """
+        Returns:
+            DataLoader: Validation dataloader
+        """
+        return DataLoader(
+            self.val,
+            collate_fn=self._collate_fn(),
+            batch_size=self.eval_batch_size,
+            drop_last=True,
+        )
+
+    def _collate_fn(self):
+        """Helper function to merge a list of samples into a batch of Tensors"""
+
+        def _collate(examples):
+            return self.tokenizer.pad(examples, return_tensors="pt")
```

Comment:
what does this do? I thought the padding was done directly when you tokenize the dataset earluer?

Reply:
This basically padded the within the batch. We don't really want to always pad to the max length as it will lead to big inputs.

Reply:
I'll set the tokenizer to never pad though, I'm seeing it's in "longest" mode now